chart/templates/_helpers.yaml

```diff
@@ -233,27 +233,30 @@ If release name contains chart name it will be used as a full name.
       value: "false"
     {{- end }}
     {{ else if .Values.dags.gitSync.credentialsSecret }}
+    {{- if hasPrefix .Values.images.gitSync.tag "v3" }}
```

Comment:
We cant determine the version based on the tag. It's possible for users to retag these images with a different scheme (e.g. calver).

Reply:
I also thought about that case, but the only alternative that came to my mind is to introduce a new key to the values.

What do you suggest to do here?

Reply:
My proposal would be something like that:

```yaml
dags:
  gitSync:
    enabled: true
    gitVersion: v4
```

But I would really try to avoid that to be honest.

Reply:
So what you suggest is that the key expected in the secret is `GIT_SYNC_X` and it will add envs for both versions of gitSync?

Reply:
Well, that's how we should have done it :). #34731 shouldn't have introduced a new key. Not sure how we should reconcile it now though...

Reply:
I change the logic to what you suggested. 

I'm not sure how you like to move on with this. I would says this is not a breaking change, since users already had to have both keys in the secret.

Please let me know when you made a decision. :)

Reply:
I think it's the right thing to do, but we do still need a newsfragment for it.